### PR TITLE
Invoke optional controller $onInit function

### PR DESCRIPTION
**Pull request by DougKeller**
https://docs.angularjs.org/guide/component#component-based-application-architecture

1.6.x introduced many mechanisms to help AngularJS 1.X projects begin conforming with Angular 2+'s component-based structure. Referencing the link above:

>Components have a well-defined lifecycle: Each component can implement "lifecycle hooks". These are methods that will be called at certain points in the life of the component.

One of the most useful hooks defined is the `$onInit` hook, which is run after the element has been constructed. An additional perk of this hook is that it can be tested individually without having to mock out any setup the controller has to do once it has been initialized.

Because AngularModalService manually initializes the modal's controller, this lifecycle hook is never invoked. This pull request checks that the initialized controller defines an `$onInit` function, and if it does, calls it before resolving the modal.

### CODE REVIEW

File: src/angular-modal-service.js [hunk 1]
```diff
@@ -183,6 +183,10 @@ module.provider('ModalService', function ModalServiceProvider() {
                             modal.close = closeDeferred.promise;
                             modal.closed = closedDeferred.promise;
 
+                            if (angular.isFunction(modal.controller.$onInit)) {
```
Comment (dwmkerr) [inline src/angular-modal-service.js]: This is excellent - but can I suggest a comment here to note that the `$onInit` function is only present in Angular 1.7, so we must check for it before firing it? Might make it clearer for anyone checking this later!

**Comment by dwmkerr** [comment]
Awesome! One minor comment, otherwise looks great! Thanks for the tests too.

**Comment by DougKeller** [comment]
Added that comment. Let me know if there's anything else - otherwise thanks for reviewing this quickly!

Also, this PR addresses #173 so that can be resolved once this is merged.

**Comment by dwmkerr** [comment]
Thanks so much for this @DougKeller! I've just released now, version `0.15.1`. Sorry for the delay, busy week! I'll check the other PR and Issue you mentioned and close them too. Cheers!

**Pull request merged**